xarray/core/dataset.py

```diff
@@ -4533,14 +4533,37 @@ def _to_dataframe(self, ordered_dims):
         index = self.coords.to_index(ordered_dims)
         return pd.DataFrame(dict(zip(columns, data)), index=index)
 
-    def to_dataframe(self):
+    def to_dataframe(self, dim_order: List[Hashable] = None):
         """Convert this dataset into a pandas.DataFrame.
 
         Non-index variables in this dataset form the columns of the
         DataFrame. The DataFrame is be indexed by the Cartesian product of
         this dataset's indices.
+
+        Parameters
+        ----------
+        dim_order : list, optional
+            Hierarchical dimension order for the resulting dataframe. All
+            arrays are transposed to this order and then written out as flat
+            vectors in contiguous order, so the last dimension in this list
+            will be contiguous in the resulting DataFrame. This has a major
+            influence on which operations are efficient on the resulting dask
+            dataframe.
+
+            If provided, must include all dimensions on this dataset. By
+            default, dimensions are sorted alphabetically.
+
         """
-        return self._to_dataframe(self.dims)
+
+        if dim_order is None:
+            dim_order = list(self.dims)
+        elif set(dim_order) != set(self.dims):
+            raise ValueError(
+                "dim_order {} does not match the set of dimensions on this "
+                "Dataset: {}".format(dim_order, list(self.dims))
+            )
```

Comment:
Maybe we could make a small helper method `_normalize_dim_order()` that we could use both here and in `to_dask_dataframe`?